string string_msg
int64 int64_msg
uint64 uint64_msg
float64 float64_msg
bool bool_msg